codegen/templates/endpoint.tmpl

```diff
@@ -50,51 +50,50 @@ import (
 
 // {{$handlerName}} is the handler for "{{.HTTPPath}}"
 type {{$handlerName}} struct {
-	Clients *module.ClientDependencies
+	Clients  *module.ClientDependencies
+	endpoint *zanzibar.RouterEndpoint
 }
 
 // New{{$handlerName}} creates a handler
-func New{{$handlerName}}(
-	gateway *zanzibar.Gateway,
-	deps *module.Dependencies,
-) *{{$handlerName}} {
-	return &{{$handlerName}}{
+func New{{$handlerName}}(deps *module.Dependencies) *{{$handlerName}} {
+	handler := &{{$handlerName}}{
 		Clients: deps.Client,
 	}
+	handler.endpoint = zanzibar.NewRouterEndpoint(
+		deps.Default.Logger, deps.Default.Scope,
+		"{{$endpointId}}", "{{$handleId}}",
+		{{ if len $middlewares | ne 0 -}}
+		zanzibar.NewStack([]zanzibar.MiddlewareHandle{
+			{{range $idx, $middleware := $middlewares -}}
+			{{$middleware.Name}}.NewMiddleWare(
+				deps.Default.Logger, deps.Default.Scope,
```

Comment:
This breaking change to the middleware api is going to be really annoying.

Let's not do it and continue to pass in `*zanzibar.Gateway` instead.

Reply:
Sure. I can change it back to `*zanzibar.Gateway`.

Reply:
Or I can pass `gateway` and `deps` to all module initializers.